assets/js/components/ErrorNotice.test.js

```diff
@@ -0,0 +1,243 @@
+					message: 'Test error message',
+					data: {
+						reason: ERROR_REASON_INSUFFICIENT_PERMISSIONS,
+					},
+					selectorData: {
+						args: [],
+						name: 'getAccountID',
+						storeName: MODULES_ANALYTICS,
+					},
+				} }
+			/>,
+			{
+				registry,
+			}
+		);
+
+		expect( queryByText( /retry/i ) ).not.toBeInTheDocument();
+	} );
+
+	it( 'should not render the `Retry` button if the error reason is `ERROR_CODE_MISSING_REQUIRED_SCOPE`', () => {
+		const { queryByText } = render(
+			<ErrorNotice
+				error={ {
+					code: ERROR_CODE_MISSING_REQUIRED_SCOPE,
+					message: 'Test error message',
+					data: {
+						reason: '',
+					},
+					selectorData: {
+						args: [],
+						name: 'getAccountID',
+						storeName: MODULES_ANALYTICS,
+					},
+				} }
+			/>,
+			{
+				registry,
+			}
+		);
+
+		expect( queryByText( /retry/i ) ).not.toBeInTheDocument();
+	} );
+
+	it( 'should not render the `Retry` button if the error is an auth error', () => {
+		const { queryByText } = render(
+			<ErrorNotice
+				error={ {
+					code: 'test-error-code',
+					message: 'Test error message',
+					data: {
+						reason: '',
+						reconnectURL: 'example.com',
+					},
+					selectorData: {
+						args: [],
+						name: 'getAccountID',
+						storeName: MODULES_ANALYTICS,
+					},
+				} }
+			/>,
+			{
+				registry,
+			}
+		);
+
+		expect( queryByText( /retry/i ) ).not.toBeInTheDocument();
+	} );
+
+	it( 'should render the `Retry` button if the error is retryable', () => {
+		const { queryByText } = render(
+			<ErrorNotice
+				error={ {
+					code: 'test-error-code',
+					message: 'Test error message',
+					data: {
+						reason: '',
+					},
+					selectorData: {
+						args: [
+							{
+								dimensions: [ 'ga:date' ],
+								metrics: [ { expression: 'ga:users' } ],
+								startDate: '2020-08-11',
+								endDate: '2020-09-07',
+							},
+						],
+						name: 'getReport',
+						storeName: MODULES_ANALYTICS,
+					},
+				} }
+			/>,
+			{
+				registry,
+			}
+		);
+
+		expect( queryByText( /retry/i ) ).toBeInTheDocument();
+	} );
+
+	it( 'should dispatch the `invalidateResolution` if the error is retryable', () => {
+		const { queryByText, getByRole } = render(
+			<ErrorNotice
+				error={ {
+					code: 'test-error-code',
+					message: 'Test error message',
+					data: {
+						reason: '',
+					},
+					selectorData: {
+						args: [
+							{
+								dimensions: [ 'ga:date' ],
+								metrics: [ { expression: 'ga:users' } ],
+								startDate: '2020-08-11',
+								endDate: '2020-09-07',
+							},
+						],
+						name: 'getReport',
+						storeName: MODULES_ANALYTICS,
+					},
+				} }
+			/>,
+			{
+				registry,
+			}
+		);
+
+		expect( queryByText( /retry/i ) ).toBeInTheDocument();
+
+		fireEvent.click( getByRole( 'button', { name: /retry/i } ) );
+
+		expect( invalidateResolutionSpy ).toHaveBeenCalledTimes( 1 );
+	} );
+
+	it( 'should not render the retry button if the error is not retryable', () => {
```

Comment:
What is this test testing specifically? There are tests for the other conditions in `isErrorRetryable` above, so this one seems to be a duplicate with less specific test spec naming.

If anything, we should add a test for when `error?.selectorData?.storeName` is false-y, since we aren't testing for that in these tests but it's in the function.